lib/engine/ets.ex

```diff
@@ -0,0 +1,34 @@
+defmodule Engine.ETS do
```

Comment:
Also thought about naming this module `Engine.Cache` since ETS is acting as our cache.

Reply:
Hmm yeah interesting thought - it's true that ETS essentially acts as our cache, although I'm not sure how I feel about abstracting it as such. It could help the cleanliness of some of the engine logic if we abstracted away the direct calls to `:ets` I suppose. Curious to see if @panentheos has thoughts

Reply:
Hmm yeah interesting thought - it's true that ETS essentially acts as our cache, although I'm not sure how I feel about abstracting it as such. It could help the cleanliness of some of the engine logic if we abstracted away the direct calls to :ets I suppose. Curious to see if @panentheos has thoughts